### PR TITLE
#561 Implement working functionality for the GET /curriculum/:curriculumAssessmentId route

### LINKED ISSUE #561: Implement functionality for the GET /curriculum/:curriculumAssessmentId route and accompanying service file functions
### Describe the Feature / Enhancement

The `GET /curriculum/:curriculumAssessmentId` route needs to be implemented and fully tested. The `GET /curriculum/:curriculumAssessmentId` route gets details for a specific curriculum assessment so a facilitator can make modifications to it as necessary.

There needs to be a test coverage for the `GET /curriculum/:curriculumAssessmentId` route and the service file functions it needs. We need to test this route to ensure the code written for the test will behave as expected by the protocol we previously defined.

This issue continues the work from issue #515.

### Additional Details and Resources

This route requires the principal ID (the user ID) from the session information, this route expects a curriculum assessment ID in the URL as a URL parameter, and expects nothing in the body in the request. The response should contain the following:

#### Facilitator

```typescript
const curriculumAssessment: CurriculumAssessment = getCurriculumAssessment();

// if curriculumAssessment is null, return a NotFoundError

res.json(itemEnvelope(curriculumAssessment));
```

#### Participant

This should return an UnauthorizedError. Participants shouldn't have access to this route.

### Correct Project Selected

- [X] This issue is linked to the appropriate project

### Labels

- [X] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by kaiyolau**
## Proposed changes

This pull request close #561  which adds a route for details of a specific CurriculumAssessment. 
It created these scenarios:

- The endpoint for this route handler is '/curriculum/:curriculumAssessmentId', where `:curriculumAssessmentId` is a URL parameter that represents the ID of the `CurriculumAssessment`.

- The programId associated with the curriculumAssessmentId is retrieved using the `getProgramIdByProgramAssessmentId()` function.

- The `programRole` of the authenticated user for this program is retrieved using the `getPrincipalProgramRole()` function.

- If the `programRole` is not defined or is 'Participant', a UnauthorizedError is thrown with an appropriate error message.

- The curriculumAssessment object associated with the `curriculumAssessmentId` is retrieved using the `getCurriculumAssessment()` function.

There is also one new function created in the Service file :

`getProgramIdByProgramAssessmentId`

## Checklist

- [x] Related issue appears at beginning of pull request title with pound sign, and title describes the changes being proposed.
- [x] Related issue is linked in pull request description using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] The appropriate label has been chosen for this pull request.
- [x] The correct project has been selected for this pull request.
- [x] All commits in this branch, including merge commits, begin with the issue number and a pound sign.
- [x] Tests have been added, where appropriate; or, no tests are relevant for this pull request.
- [x] This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes.

### CODE REVIEW

File: api/src/middleware/assessmentsRouter.ts [hunk 1]
```diff
@@ -35,9 +36,47 @@ assessmentsRouter.get('/', async (req, res, next) => {
...
+        new BadRequestError(
+          `"${curriculumAssessmentIdParsed}" is not a valid submission ID.`
+        )
+      );
+      return;
+    }
+
+    const { principalId } = req.session;
+    const programId = await getProgramIdByProgramAssessmentId(
+      curriculumAssessmentIdParsed
```
Comment (seidior) [inline api/src/middleware/assessmentsRouter.ts]: I don't think I'm understanding your logic here. Why would we call a function called `getProgramIdByProgramAssessmentId` with a variable called `curriculumAssessmentIdParsed`?

File: api/src/middleware/assessmentsRouter.ts [hunk 2]
```diff
@@ -36,9 +37,47 @@ assessmentsRouter.get('/', async (req, res, next) => {
...
+    );
+    const programIdParsed = Number(programId);
+    const programRole = await getPrincipalProgramRole(
+      principalId,
+      programIdParsed
+    );
```
Comment (seidior) [inline api/src/middleware/assessmentsRouter.ts]: I... think? I'm following your logic here, but the fatal flaw here is that you can't directly get a program ID given a curriculum assessment ID.

Instead, you'll want to check L143 through L161 to see how we tackled this logic with Paryssa's route.

File: api/src/services/assessmentsService.ts [hunk 3]
```diff
@@ -884,3 +886,24 @@ export const updateProgramAssessment = async (
 ): Promise<ProgramAssessment> => {
   return;
 };
+
+export const getProgramIdByProgramAssessmentId = async (
+  AssessmentId: number
```
Comment (seidior) [inline api/src/services/assessmentsService.ts]: I don't believe we need this function at all.

**Review by seidior — CHANGES_REQUESTED** [review]
Hi Kyle! I've brought your branch up to date with the #515 branch a couple of times so that you have the latest of what we worked on in the #sequoia-room today.

I liked that you caught some variable naming errors, but it was the change to the column name that we were selecting that was causing your tests to fail.

You had selected the pull request destination to `main` instead of to the #515 branch, so your pull request had way more changed files than what was necessary. I've since updated this to have the correct destination.

I've left you some feedback on how to proceed with this ticket. Let me know if you have any questions.

**Comment by kaiyolau** [comment]
@seidior I just updated the new logic for #561

**Review by seidior — CHANGES_REQUESTED** [review]
Hi @kaiyolau! Thanks for updating the logic. I need you to now write tests for your route as well as for any service file functions you call from your route please.

**Pull request merged**